kv/db_test.go

```diff
@@ -222,8 +222,7 @@ func TestKVDBTransaction(t *testing.T) {
 			t.Errorf("expected value %q; got %q", value, gr.ValueBytes())
 		}
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
```

Comment:
This pattern is pretty common in our code; IsTrueWithin() practically demands an inline closure in an `if` statement. I don't love it but I'm not sure that any of the alternatives are better. I don't like naming the closure because when I see that pattern I immediately start looking for more than one call site. I think that inserting extra braces for scoping may be helpful in other instances but here it just muddies the waters by adding additional braces (and indentation) that must be matched up to see where the error is being used.